plugins/state_monitor.py

```diff
@@ -0,0 +1,185 @@
+"""
+Plugin to monitor state variables from the charge controller.
+"""
+
+# Standard library imports
+
+# Local imports
+import brokkr.pipeline.base
+import brokkr.pipeline.decode
+
+from notifiers.slack import SlackSender
+
+
+class StateMonitor(brokkr.pipeline.base.OutputStep):
+    """Handle notifications for changes in state variables."""
+
+    def __init__(self,
+                 method=None,
+                 power_delim=1,
+                 slack_channel=None,
+                 slack_key_file=None,
+                 **output_step_kwargs,
+                 ):
+        """
+        Handle notifications for changes in state variables.
+
+        Parameters
+        ----------
+        method : str
+            The method of how we're going to send notifications. If `None`, then
+            we'll only log them.
+        power_delim : numeric, optional
+            The delimiter between normal power and low power.
+            If power falls below this value, it is considered low
+            and a notification will be generated.
+        slack_channel : str, optional
+            If `method=='slack'`, then this is the channel in Slack to post notifications.
+        slack_key_file : str, optional
```

Comment:
```suggestion
        slack_key_file : str or pathlib.Path, optional
```

You're just passing it through to your notifier backend, which supports both (and explicitly declares such).